src/shared/machine-credential.h

```diff
@@ -9,6 +9,14 @@ typedef struct MachineCredential {
         size_t size;
 } MachineCredential;
 
-void machine_credential_free_all(MachineCredential *creds, size_t n);
-int machine_credential_set(MachineCredential **credentials, size_t *n_credentials, const char *cred_string);
-int machine_credential_load(MachineCredential **credentials, size_t *n_credentials, const char *cred_path);
+typedef struct MachineCredentialContext {
```

Comment:
A bit of bikeshedding: I would call this just `MachineCredentials`. It's a simple wrapper for an array, after all...

Reply:
I considered it, but that way we would have something like `creds->credentials` everywhere. I'd prefer to use `ctx` if possible :)